src/components/filter/Filter.js

```diff
@@ -1,4 +1,30 @@
-function Filter() {
+import { useState, useEffect } from 'react';
+import { query, collection, getDocs, where } from 'firebase/firestore';
+import { db } from '../../services/firebaseConfig';
+
+function Filter({ title, sortBy, filteredCategory }) {
+  const [categories, setCategories] = useState([]);
+
+  const fetchCategories = async (titlee) => {
+    const setsRef = collection(db, 'sets');
+    const q = titlee
+      ? query(setsRef, where('name', '==', titlee))
+      : query(setsRef);
```

Comment:
@omarbanat I have a question here, why do we get the categories only related to the set title(name) ? I mean why not return all categories every time? 
not sure, but I'm thinking of we may get one category for each time we search for a set and the value for the set itself so it meaningless, 
and did you search for equality here right ? why not for contains?

Reply:
@hayatzsala I'm fetching all sets if the user didn't enter any work in the search field otherwise, I will fetch depending on the value entered. Regarding the filter component, especially in category section, I'm allowing the user to filter depending on the founded categories from the fetched sets.

If you still have anything please ask me.